netbox/utilities/utils.py

```diff
@@ -174,3 +178,39 @@ def to_meters(length, unit):
     if unit == LENGTH_UNIT_INCH:
         return length * 0.3048 * 12
     raise ValueError("Unknown unit {}. Must be 'm', 'cm', 'ft', or 'in'.".format(unit))
+
+
+@contextmanager
+def pg_advisory_lock(string_key):
+    """
+    Acquire an exclusive database lock against an arbitrary string key
+    """
+    try:
+        lock_key = _get_lock_key(string_key)
+        cursor = connection.cursor()
+        cursor.execute(f'SELECT pg_advisory_lock({lock_key});')
+        yield
+    finally:
+        cursor.close()
+
+
+def with_pg_advisory_lock(string_key):
+    """
+    Decorator that wraps a function with pg_advisory_lock
+    """
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kw):
+            with pg_advisory_lock(string_key):
+                return func(*args, **kw)
+        return wrapper
+    return decorator
+
+
+@lru_cache()
```

Comment:
Why the cache decorator?

Reply:
Just to avoid calculating the hash for every API call. However, SHA1 is really quite fast, so maybe this isn't worth the bother? Happy to just remove it.

Reply:
I ended up removing all of this and just using bigints that are hand-created in `constants.py`. This makes it much easier to correlate the lock IDs seen in query logs with the actual code.